docs/rfcs/91-enhance-ls-command.md

```diff
@@ -0,0 +1,197 @@
+- Author: JinnyYi <github.com/JinnyYi>
+- Start Date: 2021-11-02
+- RFC PR: [beyondstorage/beyond-ctl#91](https://github.com/beyondstorage/beyond-ctl/issues/91)
+- Tracking Issue: [beyondstorage/beyond-ctl#0](https://github.com/beyondstorage/beyond-ctl/issues/0)
+
+# BCP-91: Enhance ls Command
+
+## Background
+
+The `ls` command is used to list files or directories in Linux and other Unix-based operating systems. And it is used to list objects and common prefixes under a prefix in [Amazon CLI] and [Aliyun CLI].
+
+Currently, byctl already supports the `ls` command, but the capabilities are limited. For example, it is not possible to recursively traverse a directory and retrieve summary statistics for a directory.
+
+## Proposal
+
+So I propose to add more command options for `ls` to change the way files or directories are listed in the user terminal.
+
+### Synopsis
+
+```
+byctl ls [command options] [arguments...]
+```
+
+### Options
+
+```
+-a, --all
+    list all files
+--format
+    long -l
+-l  use a long list format
+--multipart
+    list in-progress multipart uploads
+-R, --recursive
+    list subdirectories recursively
+--summarize
+    display summary information (number of files, total size in human readable format)
+```
+
+### Usage
+
+- all type
+  - -a, --all option will show files or objects and multipart upload tasks, whose file or object name starts with the specified prefix.
+  - It returns top-level subdirectory names instead of contents of the subdirectory, which in default show by short format (output only the file name).
+  - The usage also supports -l, -R and --summarize options.
+    
+- long format
+  - -l option will show the file or object with object mode, size, last modified time and name.
+  - The usage also supports -a or --multipart and -R, --summarize options.
+
+- multipart option

```

Comment:
It looks like `-a` will have simliar behavior like `--mulitpart`?

Reply:
They have different expected results. By default, only the normal object list will be displayed. `--multipart` will only list in-progress multipart uploads. `-a` will list normal objects and multipart upload tasks.

Reply:
`--multipart` looks useful, but `-a` is a bit confused to me.

We expect to add report support (so is the `--summerize`) here, how about forcing on the job and delay all those features to the further?

Reply:
That means we only support normal objects, and add multipart/append/... support later.

Reply:
Got it. I'll remove the irrelevant options.